### PR TITLE
Add usePointSize to ShaderModuleUsage

### PR DESCRIPTION
**Pull request by xuechen417**
Expose usePointSize as a ShaderModuleUsage to check whether gl_PointSize
is used.

### CODE REVIEW

File: llpc/util/llpcShaderModuleHelper.cpp [hunk 1]
```diff
@@ -96,6 +96,9 @@ ShaderModuleUsage ShaderModuleHelper::getShaderModuleUsageInfo(const BinaryData
       if (decoration == DecorationInvariant) {
         shaderModuleUsage.useInvariant = true;
       }
+      if (decoration == DecorationBuiltIn && static_cast<BuiltIn>(codePos[4]) == BuiltInPointSize) {
```
Comment (nhaehnle) [inline llpc/util/llpcShaderModuleHelper.cpp:99]: Shouldn't the index `4` here depend on whether the opcode is OpDecorate or OpMemberDecorate? If this builtin can only appear in a struct, please at least add an assertion to that effect.

**Comment by qiaojbao** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 63913ece5e6d2f9090eac539d3733c31e3066d8e
<details><summary>CTS tests (Failed: 0/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33017/69239 (47.7%)</li>
<li>Failed: 0/69239 (0.0%)</li>
<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/util/llpcShaderModuleHelper.cpp (hunk 1 above)
Comment (xuechen417) [reply llpc/util/llpcShaderModuleHelper.cpp:99]: Yes, it should depend on OpMemberDecorte. I missed the condition. Thanks!

**Review by xuechen417 — COMMENTED** [review]
V1: Correct the condition for check BuiltInPointSize.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 039612ce84e215550d07358ee71c2ee65a245dc5
<details><summary>CTS tests (Failed: 0/146820)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Timeout or crash</li>

</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — COMMENTED** [review]
Are you sure it is correct to assume that PointSize must appear as a member? glslang generates it as such, but I don't find anything in the spec that requires it, and usually built-ins can appear on plain variables.
The following SPIR-V file at least passes the validator:
```
               OpCapability Shader
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint Vertex %main "main" %pointsize
               OpSource GLSL 450
               OpName %main "main"
      OpDecorate %pointsize BuiltIn PointSize
       %void = OpTypeVoid
          %3 = OpTypeFunction %void
      %float = OpTypeFloat 32
    %v4float = OpTypeVector %float 4
       %uint = OpTypeInt 32 0
     %uint_1 = OpConstant %uint 1
%_arr_float_uint_1 = OpTypeArray %float %uint_1
%_ptr_Output_float = OpTypePointer Output %float
    %pointsize = OpVariable %_ptr_Output_float Output
        %int = OpTypeInt 32 1
      %int_1 = OpConstant %int 1
  %float_1_5 = OpConstant %float 1.5
       %main = OpFunction %void None %3
          %5 = OpLabel
               OpStore %pointsize %float_1_5
               OpReturn
               OpFunctionEnd
```

**Comment by amdrexu** [comment]
Yes, I think Nicolai is right. The members of per-vertex structure, such as gl_Position, gl_PointSize, gl_ClipDistance, gl_CullDistance, are OpenGL-style generated by GLSLang. For SPIR-V generated from HLSL, those members are not within the structure. Rather, they are independent input/output variables. We encountered such issues before when debugging DXVK-based games.

**Review by s-perron — CHANGES_REQUESTED** [review]
I had noticed the same thing as Nicolia, but forgot to submit the review.

**Comment by xuechen417** [comment]
@nhaehnle @s-perron @amdrexu Thank you for your valuable comments. I should not be thoughtless about it. I will re-work on it and take them into consideration.

File: llpc/util/llpcShaderModuleHelper.cpp [hunk 2]
```diff
@@ -96,6 +96,10 @@ ShaderModuleUsage ShaderModuleHelper::getShaderModuleUsageInfo(const BinaryData
       if (decoration == DecorationInvariant) {
         shaderModuleUsage.useInvariant = true;
       }
+      if (opCode == OpMemberDecorate && decoration == DecorationBuiltIn &&
+          static_cast<BuiltIn>(codePos[4]) == BuiltInPointSize) {
+        shaderModuleUsage.usePointSize = true;
+      }
```
[the remark this answers is not in the document]
Comment (xuechen417) [reply llpc/util/llpcShaderModuleHelper.cpp:102]: Fixed.

**Review by xuechen417 — COMMENTED** [review]
V2: Check variable and structure-type member for BuiltInPointSize

**Comment by amdvlk-admin** [comment]
### Test summary for commit 1503bb4101eb1e5a5f213370ecda3e96e9472381
<details><summary>CTS tests (Failed: 0/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33017/69239 (47.7%)</li>
<li>Failed: 0/69239 (0.0%)</li>
<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — COMMENTED** [review]
LGTM, though be careful about conflicts in merging with the version number.

**Comment by xuechen417** [comment]
V3: Resolve conflicts.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4def2fb0da73303a0bd8ac23c140401a570c5fbc
<details><summary>CTS tests (Failed: 3/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33014/69239 (47.7%)</li>
<li><details><summary>Failed: 3/69239 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.memory.pageable_allocation.random.67
Stack trace: Flake

FAILURE: dEQP-VK.synchronization2.signal_order.shared_binary_semaphore.write_image_vertex_read_image_tess_control.image_128x128_r16_uint_opaque_fd
Stack trace: Flake

FAILURE: dEQP-VK.ycbcr.plane_view.image_view.g8_b8_r8_3plane_420_unorm_plane_2_compatible_format_r8_uint
Stack trace: Flake
```
</p></details></li>

<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4def2fb0da73303a0bd8ac23c140401a570c5fbc
<details><summary>CTS tests (Failed: 1/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33016/69239 (47.7%)</li>
<li><details><summary>Failed: 1/69239 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.subgroups.clustered.compute.subgroupclusteredmin_dvec3
Stack trace: Flake
```
</p></details></li>

<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4def2fb0da73303a0bd8ac23c140401a570c5fbc
<details><summary>CTS tests (Failed: 1/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33016/69239 (47.7%)</li>
<li><details><summary>Failed: 1/69239 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.sparse_resources.image_sparse_memory_aliasing.2d_array.rg32i.503_137_3
Stack trace: Flake
```
</p></details></li>

<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdrexu** [comment]
@xuechen417 New confliction :)

**Comment by amdrexu** [comment]
@s-perron Are you happy with the update?

**Review by s-perron — COMMENTED** [review]
LGTM

**Comment by xuechen417** [comment]
V4: Rebase.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 31c7a10e1f524ba6130ce023ec07df396a80a713
<details><summary>CTS tests (Failed: 0/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33017/69239 (47.7%)</li>
<li>Failed: 0/69239 (0.0%)</li>
<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Pull request merged**